src/views/modal_handler.rs

```diff
@@ -82,6 +95,7 @@ impl DisplayModal {
             // Display warning for unsaved changes
             ModalMessage::UnsavedChangesExitModal => {
                 self.show_modal = true;
+                self.is_warning = true;
                 self.modal_type = Some(ModalType::Warning {
```

Comment:
Aha, you have very similar to what I was saying - can we use that and avoid the boolean flags for show_modal, and is_warnign and is_version?